citro3d/examples/triangle.rs

```diff
@@ -136,22 +148,12 @@ fn scene_init(program: &mut shader::Program) -> (i8, C3D_Mtx, *mut libc::c_void)
             projection.assume_init()
         };
 
-        // Create the vertex buffer object
-        let vbo_data: *mut Vertex = ctru_sys::linearAlloc(
-            std::mem::size_of_val(&VERTICES)
-                .try_into()
-                .expect("size fits in u32"),
-        )
-        .cast();
-
-        vbo_data.copy_from(VERTICES.as_ptr(), VERTICES.len());
-
         // Configure buffers
         let buf_info = citro3d_sys::C3D_GetBufInfo();
         citro3d_sys::BufInfo_Init(buf_info);
         citro3d_sys::BufInfo_Add(
             buf_info,
-            vbo_data.cast(),
+            vbo_data.as_ptr().cast(),
```

Comment:
It would be cool to get a safe wrapper for this. Currently it's taking a reference to the buffer, but ownership is returned at the end of the function, so moving the buffer (ex. if the vec gets resized) would cause a memory issue.

Reply:
#14